lib/pipe.js

```diff
@@ -0,0 +1,40 @@
+import {Writable} from 'node:stream';
+import {createWriteStream} from 'node:fs';
+import {ChildProcess} from 'node:child_process';
+
+const isWritableStream = target => target instanceof Writable;
+
+const isExecaChildProcess = target => target instanceof ChildProcess && isWritableStream(target.stdin) && typeof target.then === 'function';
```

Comment:
`instanceof` is not cross-realm safe for `Writable` and `ChildProcess`.
However, checking individual properties for existence seems brittle.

We could either use a `Symbol` key on Execa child processes, or a global `WeakSet`. However, that would not work for `instanceof Writable`.

So I opted for only using `instanceof`, but I am not sure whether this is a good choice.